TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/apiservers/ZomatoRequest.java

```diff
@@ -0,0 +1,158 @@
+
+import com.fbu.thefoodienetwork.BuildConfig;
+import com.fbu.thefoodienetwork.keys.ZomatoKeys;
+import com.fbu.thefoodienetwork.models.Location;
+import com.fbu.thefoodienetwork.models.Restaurant;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class ZomatoRequest {
+    //endpoints
+    public static final String LOCATIONS = "locations";
+    public static final String LOCATION_DETAILS = "location_details";
+    public static final String SEARCH = "search";
+    public static final String GEO_CODE = "geocode";
+    public static final String RESTAURANT = "restaurant";
+    private static final String TAG = "ZomatoRequest";
+    private static final String API_KEY = BuildConfig.ZOMATO_KEY;
+    private static final String BASE_URL = "https://developers.zomato.com/api/v2.1/";
+
+
+    private OkHttpClient client = new OkHttpClient();
+    private HttpUrl.Builder urlBuilder;
+
+    public List<Location> getLocations(String query) {
+        final List<Location> locationList = new ArrayList<>();
+        urlBuilder = HttpUrl.parse(BASE_URL + LOCATIONS).newBuilder();
+        urlBuilder.addQueryParameter(ZomatoKeys.API_KEY, API_KEY);
+        urlBuilder.addQueryParameter(ZomatoKeys.COUNT, "10"); ////max number of results to display
+        urlBuilder.addQueryParameter(ZomatoKeys.QUERY, query); //search keyword
+        String url = urlBuilder.build().toString();
+
+        Request request = new Request.Builder()
+                .url(url)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onResponse(Call call, final Response response) throws IOException {
+                if (response.isSuccessful()) {
+                    String responseData = response.body().string();
+                    try {
+                        JSONObject json = new JSONObject(responseData);
+                        JSONArray locations = json.getJSONArray(ZomatoKeys.LOCATION_SUGGESTIONS);
+                        locationList.addAll(Location.fromJsonArray(locations));
+                        Log.i(TAG, locationList.toString());
+                    } catch (JSONException e) {
+                        Log.i(TAG, "error: " + e);
+                    }
+                }
+            }
+
+            @Override
+            public void onFailure(@NotNull Call call, @NotNull IOException e) {
+                Log.d(TAG, "onFaliure");
+            }
+        });
+        return locationList;
+    }
+
+    public void getLocationByGeoPoint(double lat, double lon, final GeoLocationCallbacks callbacks) {
+        final Location[] location = new Location[1];
```

Comment:
Does this need to be an array?  I think I see what you're trying to do, however, `location` is not used outside of the `onResponse` callback below, so it doesn't really seem like this variable is really needed.  However, if you did need the retrieved `Location` outside of the callback, you could use `AtomicReference<Location>` (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html), which reads a bit better.